internal/pkg/convert/graph.go

```diff
@@ -107,15 +107,17 @@ func (graph *GraphLLB) buildBaseImages() {
 }
 
 func (graph *GraphLLB) buildChecksummer() {
+	// use reproducible stagex images to get `sha512sum` binary
+	const stagexRelease = "sx2024.12.0"
+
 	graph.Checksummer = llb.Image(
-		constants.DefaultBaseImage,
+		"stagex/core-musl:"+stagexRelease+"@sha256:14fd2adc660de2d3dc319e0326d59633888e9ba42214cba1e09cfad616f982fe",
```

Comment:
maybe use `stagex/core-busybox` instead of dynamically linked coreutils? Also keep in mind it's amd64 only and would require qemu to run on ARM

Reply:
oh yeah... lack of arm64 might be a showstopper here. 

I hate busybox now, but it's an option of course.

Reply:
Perhaps in our case bldr itself executes on the primary node (amd64 where action itself runs), but these commands seem to be injected into the build flow scheduled on all nodes

Reply:
this step is run per-arch, sources might be different on each arch

Reply:
resolved now with new image